threshold_tools_event_types.ipynb

```diff
@@ -0,0 +1,505 @@
+    "\n",
+    "To perform the example analyses provided later in this notebook, choose hourly temperature data in degrees Fahrenheit, for SSP 3-7.0. Choose a smaller area subset (such as one county), and take an area average to make the example analysis faster.\n",
+    "\n",
+    "To learn more about the data available on the Analytics Engine, see our [data catalog](https://analytics.cal-adapt.org/data/). The *getting_started.ipynb* notebook contains additional explanations of the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "app.select()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Run `app.retrieve()` to load the data selected above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hourly_data = app.retrieve()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the following analysis, we will work with a subset of the data from just one model simulation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hourly_data_subset = hourly_data.isel(simulation=0, scenario=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To speed up the following computations, load the selected subset of data into memory. This step may take a few minutes to compute."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hourly_data_subset = app.load(hourly_data_subset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='transform'></a>\n",
+    "## Step 2: Get block maxima series for different event types\n",
+    "\n",
+    "To facilitate custom definitions of different types of extreme events, there are optional keyword arguments to the `get_block_maxima` function to pull the maximum annual value. The keyword options are:\n",
+    "- `duration`: how long an event continuously lasts in hours (see example 2)\n",
+    "- `groupby` and  `grouped_duration`: work together to define multi-day events (see examples 3 and 4)\n",
+    "\n",
+    "The following four examples show how to use these options to construct different types of events of interest. See section 3 for further analysis and discussion of these events."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example 1: hottest hour in each year\n",
+    "The basic use case of `get_block_maxima` pulls the maximum value in each year of data, shown here:"
```

Comment:
so the estimated ~23 is the average ESS per year (or per block if block_size is different than 1 year). So it's not about the number of years, but the effective sample size within each year. This has to do with the amount of autocorrelation and the type of event (if you define a 10-day long event, the sample size in each year is smaller than an hour long event). 

I hadn't run into a case of getting an average ESS <25 for example 1 though. Do you remember what your other selections were? this was hourly data right?

Reply:
Hourly, temperature in F, 1980-2100 for SSP 3-7.0 for a county w/ area average for Amador county

Using average ESS across all years is an interesting diagnostic - temporal autocorrelation of temperature at least is not necessarily constant through end of century (depending on how you're calculating it), so I'm curious whether there is any directional trend in ESS across years that might bias that average?

Reply:
I played around with this, and you're totally right-- depending which county you select, and whether or not you include historical climate data in the selected data, you can end up with the "too small ESS" warning in the first two examples, as you found. I think this is tricky because we don't know when the user will first encounter the error, so for now, I'm inclined to make "Sacramento County" the official focus of this example notebook so that the explanatory text for ESS stays at the 3rd example, where the warning will first be thrown. Also, depending on which area you use, the example return value of 105F (in the Step 3 EVA section) may not be an informative value to use (might end up with very large return periods), but it is a good value to use for Sacramento. I added a note addressing this, and added a comment in the selection section to use Sacramento county for the initial exploration in this notebook.

Reply:
I think that's a reasonable plan! It makes sense to not want to bog down the reader w/ all the details right away